Parsedown.php

```diff
@@ -1485,10 +1503,80 @@ function parse($text)
         return $markup;
     }
 
+    protected function sanitiseElement(array $Element)
+    {
+        static $badAttributeChars = "\"'= \t\n\r\0\x0B";
+        static $safeUrlNameToAtt  = array(
+            'a'   => 'href',
+            'img' => 'src',
+        );
+
+        if (isset($safeUrlNameToAtt[$Element['name']]))
+        {
+            $Element = $this->filterUnsafeUrlInAttribute($Element, $safeUrlNameToAtt[$Element['name']]);
+        }
+
+        if ( ! empty($Element['attributes']))
+        {
+            foreach ($Element['attributes'] as $att => $val)
+            {
+                # clear out nulls
+                if ($val === null)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+                # filter out badly parsed attribute
+                elseif (strpbrk($att, $badAttributeChars) !== false)
+                {
+                    unset($Element['attributes'][$att]);
+                }
+            }
+
+            $onEventAttributeKeys = preg_grep('/^on/i', array_keys($Element['attributes']));
+
+            foreach ($onEventAttributeKeys as $att)
+            {
+                unset($Element['attributes'][$att]);
+            }
+        }
+
+        return $Element;
+    }
+
+    protected function filterUnsafeUrlInAttribute(array $Element, $attribute)
+    {
+        if ($this->safeLinksEnabled)
+        {
+            $safe = false;
+
+            foreach ($this->safeLinksWhitelist as $scheme)
+            {
+                if (stripos($Element['attributes'][$attribute], $scheme) === 0)
```

Comment:
I'm happy to switch to something you can point to that works 😉

Generally speaking, whitelists are safer than blacklists (i.e. try and figure out what to allow, rather than what to block). So even if the HTML decode method I suggested above seems to work, I'd personally still opt to use a whitelist over it. (since it's just a fancy blacklist).